File: CSC/wtfc24/A1/MyZip.java
import java.util.LinkedList;

class MyZip {
	public static LinkedList myZipRec(LinkedList xs, LinkedList ys){
		if (xs.size() == ys.size()){
			if (xs.size() == 0){
				return xs;
			} else {
				LinkedList result = new LinkedList();
				LinkedList temp = new LinkedList();
				temp.add(xs.pop());
				temp.add(ys.pop());
				result.add(temp);
				result.addAll(myZipRec(xs, ys));
				return result;
			}
		}
		throw new Error();
	}

	public static LinkedList myZip(LinkedList xs, LinkedList ys){
		if (xs.size() == ys.size()){
			LinkedList result = new LinkedList();
			for (int i=0; i<xs.size()+1; i++){			
				LinkedList temp = new LinkedList();
				temp.add(xs.pop());
				temp.add(ys.pop());
				result.add(temp);
			}
			return result;
		}
		throw new Error();
	}
}

File: CSC/wtfc24/A1/myzip.py
'''(my-zip-rec xs ys) -> list
xs, ys: lists of same length
return a list of pairs of corresponding elements from xs and ys
e.g., my_zip([1,2],[3,4]) ==> [[1,3], [2,4]]
implement this function recursively'''
def my_zip_rec(xs, ys):
	if len(xs) == len(ys):
		if xs == []:
			return []
		else:
			final = [[xs[0], ys[0]]]
			final.extend(my_zip_rec(xs[1:], ys[1:]))
			return final
	else:
		return "Error"

'''non-recursive version'''
def my_zip(xs, ys):
	if len(xs) == len(ys):
		final = []
		for i in range(len(xs)):
			final.append([xs[i], ys[i]])
		return final
	else:
		return "Error"